controller/work_item_link.go

```diff
@@ -513,3 +518,14 @@ func ConvertLinkToModel(appLink app.WorkItemLinkSingle) (*link.WorkItemLink, err
 
 	return &modelLink, nil
 }
+
+func ValidateSingleParent(ctx *workItemLinkContext, targetID uint64, linkTypeID uuid.UUID) error {
+	result := link.WorkItemLink{}
+	var db *gorm.DB
+	db = db.Where("link_type_id=? AND target_id=?", linkTypeID, targetID).Find(&result)
```

Comment:
@kwk Can you please ellaborate this? What do you mean by **duplicate** ?
This issue talks about disabling **multiple** parents of a single WI.

My logic is:
Consider 3 workitems -> 1, 2, 3
To disallow multiple parents of a workitem means:
-> I can add 1 as a parent of 3 
BUT 
-> I cannot add 2 as a parent of 3 (unless i delete the first link)

Consider this:
```
             link_type_id             | source_id | target_id 
--------------------------------------+-----------+-----------
 b1f6404a-a869-422e-8cec-aa4beb6f0693 |         1 |         3
 b1f6404a-a869-422e-8cec-aa4beb6f0693 |         2 |         3
```
To disallow **multiple** parents, I have added constraint such that the `work_item_links` table should not allow  row 2. 
Am i missing something?